### PR TITLE
update UI, TakeScreenShot (API)

### PR DESCRIPTION
**Pull request by laiglinne-ff**

### CODE REVIEW

**Comment by laiglinne-ff** [comment]
@RyuaNerin Yeah Awesome Flying Spagetti

**Comment by laiglinne-ff** [comment]
I Hate Lambda!!!!!!!!!!!!!!!!!@#!@!!

**Comment by hibiyasleep** [comment]
Please remove ACTColumnAdder - it should exist as an individual plugin, not here.

File: OverlayPlugin.Common/IOverlay.cs [hunk 1]
```diff
@@ -63,5 +58,11 @@ public interface IOverlay : IDisposable
         /// </summary>
         /// <param name="message">A string message created by the plugin javascript.</param>
         void OverlayMessage(string message);
+
+        /// <summary>
+        /// Take Overlay Renderer PNG Image (Offscreen?)
+        /// </summary>
+        /// <param name="path"></param>
+        void TakeScreenShot(ScreenshotConfig config);
```
Comment (hibiyasleep) [inline OverlayPlugin.Common/IOverlay.cs]: Can you replace all `ScreenShot` with `Screenshot`? it's actually one word.

File: OverlayPlugin.Core/Localization.cs [hunk 2]
```diff
@@ -65,13 +62,16 @@ static Localization()
...
-            dict[TextItem.ToggleVisible, "ko"] = "표시 / 숨기기 토글";
+            dict[TextItem.ToggleVisible, "ko"] = "표시 / 숨기기";
             dict[TextItem.ToggleClickthru, ""] = "Toggle clickthru";
             dict[TextItem.ToggleClickthru, "ja"] = "クリック透過の切り替え";
-            dict[TextItem.ToggleClickthru, "ko"] = "Toggle clickthru";
+            dict[TextItem.ToggleClickthru, "ko"] = "마우스 통과 끄기 / 켜기";
             dict[TextItem.ToggleLock, ""] = "Toggle lock";
             dict[TextItem.ToggleLock, "ja"] = "移動/リサイズ制限の切り替え";
-            dict[TextItem.ToggleLock, "ko"] = "Toggle lock";
+            dict[TextItem.ToggleLock, "ko"] = "위치 잠구기 끄기 / 켜기";
```
Comment (hibiyasleep) [inline OverlayPlugin.Core/Localization.cs]: 잠구기 -> 잠그기

File: OverlayPlugin.Core/Overlays/LabelOverlay/LabelOverlayConfigPanel.cs [hunk 3]
```diff
@@ -175,6 +158,7 @@ private void textBoxGlobalHotkey_KeyDown(object sender, KeyEventArgs e)
 
         private void checkLock_CheckedChanged(object sender, EventArgs e)
         {
+            if (this.config.Url == "" || this.config.Text == "") return;
```
Comment (hibiyasleep) [inline OverlayPlugin.Core/Overlays/LabelOverlay/LabelOverlayConfigPanel.cs]: Checkbox config should can be changed even there's no URL or text specified, please fix this.

File: OverlayPlugin.Core/Overlays/LogParseOverlay/LogParseOverlay.cs [hunk 4]
```diff
@@ -92,7 +88,7 @@ protected override void Update()
                 this.prevEndDateTime = ActGlobals.oFormActMain.ActiveZone.ActiveEncounter.EndTime;
                 this.prevEncounterActive = ActGlobals.oFormActMain.ActiveZone.ActiveEncounter.Active;
 
-                var updateScript = CreateEventDispatcherScript();
+                var updateScript = CreateEventDispatcherScript();
```
Comment (hibiyasleep) [inline OverlayPlugin.Core/Overlays/LogParseOverlay/LogParseOverlay.cs]: this file has some trailing spaces.

File: OverlayPlugin.Core/Overlays/MiniParseOverlay/MiniParseOverlay.cs [hunk 5]
```diff
@@ -18,14 +16,20 @@ public partial class MiniParseOverlay : OverlayBase<MiniParseOverlayConfig>
...
+        {
+            get { return updateStringCacheExpireInterval; }
+            set { updateStringCacheExpireInterval = value; }
+        }
 
         public MiniParseOverlay(MiniParseOverlayConfig config)
             : base(config, config.Name)
         {
             ActGlobals.oFormActMain.BeforeLogLineRead += LogLineReader;
+            ActGlobals.oFormActMain.OnCombatStart += this.OFormActMain_OnCombatStart; ;
```
Comment (hibiyasleep) [inline OverlayPlugin.Core/Overlays/MiniParseOverlay/MiniParseOverlay.cs]: This line has 2 semicolons.

File: OverlayPlugin.Core/PluginMain.cs [hunk 6]
```diff
@@ -132,6 +90,80 @@ public void InitPlugin(TabPage pluginScreenSpace, Label pluginStatusText)
...
+                var targetOverlay = Overlays.FirstOrDefault(x => x.Name == e.Target);
+                if (targetOverlay != null)
+                {
+                    targetOverlay.OverlayMessage(e.Message);
+                }
+            });
+        }
+
+        /// <summary>
+        /// Take Screenshot target overlay (API: window.overlayApi.takeScreenShot( window.overlayApi.takeScreenShot))
```
Comment (hibiyasleep) [inline OverlayPlugin.Core/PluginMain.cs]: `window.overlayPluginApi.takeScreenshot(window.overlayPluginApi.overlayName)`?

File: OverlayPlugin.Core/ScreenShot.cs [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+                }
+
+                if (config.AutoClipping)
+                    bitmap = AutoClipping(bitmap);
+
+            }
+            catch (Exception ex)
+            {
+                bitmap?.Dispose();
+                MessageBox.Show(ex.ToString());
```
Comment (hibiyasleep) [inline OverlayPlugin.Core/ScreenShot.cs]: Can't this be `Log`ged?

File: Thirdparty/CefBinary_x86/resources/label.html [hunk 8]
```diff
@@ -0,0 +1,109 @@
+﻿<html>
```
Comment (hibiyasleep) [inline Thirdparty/CefBinary_x86/resources/label.html]: This file seems duplicated by accident, please remove this.

**Comment by RyuaNerin** [comment]
👍

File: OverlayPlugin.Core/PluginMain.cs [hunk 9]
```diff
@@ -126,7 +126,7 @@ private void OverlayMessageEvent(object sender, SendMessageEventArgs e)
         }
 
         /// <summary>
-        /// Take Screenshot target overlay (API: window.overlayApi.takeScreenShot( window.overlayApi.takeScreenShot))
+        /// Take Screenshot target overlay (on HTML: window.overlayApi.takeScreenShot( window.overlayApi.overlayName ))
```
Comment (laiglinne-ff) [inline OverlayPlugin.Core/PluginMain.cs]: I think, after create developer guide. maybe

File: OverlayPlugin.Core/PluginConfig.cs [hunk 10]
```diff
@@ -331,21 +331,29 @@ public static PluginConfig LoadXml(string pluginDirectory, string path)
         /// <param name="pluginDirectory"></param>
         public void SetDefaultOverlayConfigs(string pluginDirectory)
         {
-            var miniparseOverlayConfig = new MiniParseOverlayConfig(DefaultMiniParseOverlayName);
-            miniparseOverlayConfig.Position = new Point(20, 20);
-            miniparseOverlayConfig.Size = new Size(500, 300);
-            miniparseOverlayConfig.Url = new Uri(Path.Combine(pluginDirectory, "resources", "miniparse.html")).ToString(); 
+            var KagerouDefaultOverlay = new MiniParseOverlayConfig("Kagerou Overlay");
```
Comment (laiglinne-ff) [inline OverlayPlugin.Core/PluginConfig.cs]: Modern Overlays

**Comment by laiglinne-ff** [comment]
image
swag;

**Comment by laiglinne-ff** [comment]
Ugh Rebased

File: OverlayPlugin.Core/PluginConfig.cs (hunk 10 above)
Comment (hibiyasleep) [reply OverlayPlugin.Core/PluginConfig.cs]: please don't do this - I'll leave OverlayPlugin independent in any ways.

**Pull request closed without merging**